src/plugins-registry.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { loadModule } from './dom-utils.js';
+
+export default class PluginsRegistry {
+  #plugins;
+
+  static parsePluginParams(id, config) {
+    const pluginId = !config
+      ? id.split('/').splice(id.endsWith('/') ? -2 : -1, 1)[0].replace(/\.js/, '')
+      : id;
+    const pluginConfig = {
+      load: 'eager',
```

Comment:
Plugins would load in the `eager` phase by default. should we set this to `lazy` so we keep the best performance by default, and require explicit opt-in to `eager` for those plugins that would need immediate instrumentation?